flexget/plugins/clients/flood.py

```diff
@@ -0,0 +1,446 @@
+            url: http://localhost:3000  # Required. Url for Flexget to connect to Flood.
+            username: flexget           # Required. Username for authentication with Flood.
+            password: password          # Required. Password for authentication with Flood.
+            action: add                 # Required. The action to perform. Can be 'add', 'remove', 'delete', 'start', or 'stop'.
+            path: /downloads            # If the action is set to 'add', the path is relative to the Flood download directory.
+            tags: [ 'Flexget' ]         # If the action is set to 'add', the tags to add to the torrent.
+    """
+
+    schema = {
+        'type': 'object',
+        'properties': {
+            'url': {'type': 'string'},
+            'username': {'type': 'string'},
+            'password': {'type': 'string'},
+            'action': {'type': 'string', 'enum': ['add', 'remove', 'delete', 'start', 'stop']},
+            'path': {'type': 'string'},
+            'tags': {'type': 'array', 'items': {'type': 'string'}},
+        },
+        'additionalProperties': False,
+        'required': ['url', 'username', 'password', 'action'],
+    }
+
+    def _add_entry_file(
+        self, task: Task, config: dict, entry: Entry, destination: str, tags: list, start: bool
+    ) -> None:
+        """
+        Process an entry to add to Flood by file.
+        """
+
+        if not os.path.exists(entry['file']):
+            return entry.fail('Temp file not found.')
+
+        with open(entry['file'], 'rb') as fs:
+            encoded_file = base64.b64encode(fs.read()).decode('utf-8')
+
+            try:
+                FloodClient.add_torrent_files(
+                    task, config, [encoded_file], destination, tags, start
+                )
+            except PluginError as e:
+                return entry.fail(e)
+
+    def _add_entry_url(
+        self, task: Task, config: dict, entry: Entry, destination: str, tags: list, start: bool
+    ) -> None:
+        """
+        Process an entry to add to Flood by URL.
+        """
+
+        try:
+            FloodClient.add_torrent_urls(task, config, [entry['url']], destination, tags, start)
+        except PluginError as e:
+            return entry.fail(e)
+
+    def _process_accepted_entry(self, task: Task, config: dict, entry: Entry) -> None:
+        if config['action'] == 'add':
+            self.add_entry(task, config, entry)
+        elif 'flood_hash' in entry:
+            if config['action'] == 'remove':
+                FloodClient.delete_torrents(task, config, [entry['flood_hash']], False)
+            elif config['action'] == 'delete':
+                FloodClient.delete_torrents(task, config, [entry['flood_hash']], True)
+            elif config['action'] == 'start':
+                FloodClient.start_torrents(task, config, [entry['flood_hash']])
+            elif config['action'] == 'stop':
+                FloodClient.stop_torrents(task, config, [entry['flood_hash']])
```

Comment:
Maybe log a warning or throw an exception if `'flood_hash'` isn't available?

Reply:
I did originally have a check to see if the torrent exists before trying to perform any of the actions that require a hash, but Flood will instead throw a status 500, I figured that it would be picked up by L37's try-catch block

Reply:
Added a entry.fail if no `flood_hash` was found in Winter/Flexget@6f32337